modules/misc/qt/kconfig.nix

```diff
@@ -0,0 +1,77 @@
+{ config, pkgs, lib, ... }:
+
+let
+
+  cfg = config.qt.kde.settings;
+
+  toKconfVal = p: v:
+    let t = builtins.typeOf v;
+    in if t == "set" then
+      v
+    else if v == null then
+      "--delete"
+    else if t == "bool" then
+      "--type bool ${builtins.toJSON v}"
+    else
+      toString v;
+in {
+  options.qt.kde.settings = lib.mkOption {
+    type = with lib.types;
+      let
+        valueType =
+          nullOr (oneOf [ bool int float str path (attrsOf valueType) ]) // {
+            description = "KDE option value";
+          };
+      in attrsOf valueType;
+    default = { };
+    example = {
+      powermanagementprofilesrc.AC.HandleButtonEvents.lidAction = 32;
+    };
+    description = ''
+      A set of values to be modified by {command}`kwriteconfig5`.
+
+      The example value would cause the following command to run in the
+      activation script:
+
+      ``` shell
+      kwriteconfig5 --file $XDG_CONFIG_HOME/powermanagementprofilesrc \
+                    --group AC \
+                    --group HandleButtonEvents \
+                    --group lidAction \
+                    --key lidAction \
+                    32
+      ```
+
+      Note, `null` values will delete the corresponding entry instead of
+      inserting any value.
+    '';
+  };
+
+  config = lib.mkIf (cfg != { }) {
+    home.activation.kconfig = lib.hm.dag.entryAfter [ "writeBoundary" ] ''
+      source ${
+        pkgs.runCommandLocal "kwriteconfig.sh" {
+          passAsFile = [ "cfg" "jqScript" ];
+          cfg = builtins.toJSON (lib.mapAttrsRecursive toKconfVal cfg);
+          jqScript = ''
```

Comment:
I wrote it in nix with support of multiple inner groups without this `jq` sorcery, if you are interested:

https://github.com/kurnevsky/nixfiles/commit/68ed3b9dca05b1bbaab666d3913aba6b66736852#diff-580825c1977fc8552d050516c2b2eb93dcb9d0ce88dfece8bba8816d576ad290R317

Reply:
thanks, I followed your example